samples/context-app-csharp/ContextAppForDSS/ContextualDataOperation.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Akri.Mq.StateStore;
+using Akri.Mqtt.Connection;
+using Akri.Mqtt.Session;
+using ContextualDataIngestor;
+using Microsoft.Extensions.Logging;
+using System.Runtime.CompilerServices;
+using Akri.Mqtt.Models;
+
+namespace ContextAppForDSS
+{
+    internal class ContextualDataOperation
+    {
+        private IDataRetriever _dataRetriever;
+        private ILogger _logger;
+        private readonly Dictionary<string, string> _parameters;
+
+        public ContextualDataOperation(IDataRetriever dataRetriever, Dictionary<string, string> parameters, ILogger logger)
+        {
+            _parameters = parameters;
+            _dataRetriever = dataRetriever;
+            _logger = logger;
+        }
+
+        public async Task PopulateContextualDataAsync()
+        {
+            
+            // MQTT Communication
+            await using MqttSessionClient mqttClient = await SetupMqttClient();
+            IStateStoreClient stateStoreClient = new StateStoreClient(mqttClient);
```

Comment:
```suggestion
            await using IStateStoreClient stateStoreClient = new StateStoreClient(mqttClient);
```

This will automatically close and dispose this object once it leaves scope